action/protocol/staking/candidate.go

```diff
@@ -323,3 +323,37 @@ func delCandidate(sm protocol.StateManager, name address.Address) error {
 	_, err := sm.DelState(protocol.NamespaceOption(CandidateNameSpace), protocol.KeyOption(name.Bytes()))
 	return err
 }
+
+func getAllCandidates(sr protocol.StateReader) (uint64, CandidateList, error) {
+	height, iter, err := sr.States(protocol.NamespaceOption(CandidateNameSpace))
+	if errors.Cause(err) == state.ErrStateNotExist {
+		return height, nil, nil
```

Comment:
easy to cause nil ptr when your return nil value and nil err

Reply:
and this is an inconsistent behavior compare to `getCandidate` which return an error `state.ErrStateNotExist` when not found

Reply:
fixed